doc/doc-requirements.txt

```diff
@@ -1,15 +1,25 @@
+# Order matters:
+# - myst-parser depends on "sphinx>=2,<4"
+# - pydata-sphinx-theme depends on "sphinx"
+# - sphinx-copybutton depends on "sphinx>=1.8"
+#
+# Listing either pydata-sphinx-theme or sphinx-copybutton first will make the
+# myst-parser constraints on sphinx be ignored, so myst-parser should go
+# first. This is only relevant if sphinx==1.* is already installed in the
+# environment, which it is on ReadTheDocs.
+
 # documentation specific extra packages
 myst-parser
+pydata-sphinx-theme
 sphinx-copybutton
 traitlets
 pandas
 ruamel.yaml
-pydata_sphinx_theme
-sphinx>=1.7
+
 # install BinderHub dependencies. We manually list them here because some
 # dependencies (like pycurl) can't be installed on ReadTheDocs and aren't
 # needed to build the docs.
-kubernetes==9.0.*
+kubernetes
```

Comment:
I unpinned this, which is currently at 0.12.0 I think, and the latest KubeSpawner part of z2jh 0.10.0 works fine with 0.12.0.